Add Feedback mailtos to app and dashboard

tallcoleman opened the pull request:
Closed #100 

Mailto link added for dashboard and app to help users report feedback.

Email is like this:

To: bikespaceto@gmail.com

Subject: BikeSpace App Feedback

> \----------
> Please describe your feedback about app.bikespace.ca below. We welcome both positive feedback (e.g. I found x feature useful) and constructive feedback (e.g. y is broken, I wish the app did z).
> 
> Especially for constructive feedback, you can help us by letting us know:
> - Your browser and platform (e.g. Safari, iPhone)
> - If it's a bug, what steps led to the problem
> - If it's something you wish the app was able to do, what goal would that feature help you accomplish? (e.g. "I think it would help advocates if you could report issues of x type", or "I wanted to be able to do step y easier on my phone")
> 
> Thank you for taking the time to help us make the app better!
> \----------
> 
> Hi BikeSpace team,

tallcoleman committed:
Added mailto feedback link to dashboard

tallcoleman committed:
Added feedback mailto to app submission page

hulloanson commented on bikespace_dashboard/index.html:
```diff
@@ -53,6 +53,15 @@
           </li>
+          <li>
+            <a
+              href="mailto:bikespaceto@gmail.com?subject=BikeSpace%20Dashboard%20Feedback&body=----------%0APlease%20describe%20your%20feedback%20about%20dashboard.bikespace.ca%20below.%20We%20welcome%20both%20positive%20feedback%20(e.g.%20I%20found%20x%20feature%20useful)%20and%20constructive%20feedback%20(e.g.%20y%20is%20broken%2C%20I%20wish%20the%20dashboard%20did%20z).%0A%0AEspecially%20for%2 ...
```
let's use encodeURIComponent here instead? It is kind of hard to read here

tallcoleman replied:
That's a good idea... let me look that up

tallcoleman replied:
@hulloanson  done

hulloanson commented on bikespace_frontend/src/components/FeedbackMailTo.tsx:
```diff
@@ -0,0 +1,18 @@
+
+const FeedbackMailTo = () => {
+  return (
+    <a href="mailto:bikespaceto@gmail.com?subject=BikeSpace%20App%20Feedback&body=----------%0APlease%20describe%20your%20feedback%20about%20app.bikespace.ca%20below.%20We%20welcome%20both%20positive%20feedback%20(e.g.%20I%20found%20x%20feature%20useful)%20and%20constructive%20feedback%20(e.g.%20y%20is%20broken%2C%20I%20wish%20the%20app%20did%20z).%0A%0AEspecially%20for%20constructive%20feedback% ...
```
encodeURIComponent?

tallcoleman replied:
@hulloanson done

tallcoleman committed:
Changed dashboard mailto to use encodeURIComponent

Progressive enhancement - no JS will still have a working mailto
Also added an umami tracking attribute

tallcoleman committed:
Changed app mailto to use encodeURIComponent

sonalranjit commented on bikespace_frontend/src/components/FeedbackMailTo.tsx:
```diff
@@ -0,0 +1,41 @@
+
+const feedback_to = 'bikespaceto@gmail.com';
+const feedback_subject = 'BikeSpace App Feedback';
+const feedback_body = `----------
```
Could we follow the google typescript guide for camelcase for variable declarations. The linter should've picked these up 🤔 
https://google.github.io/styleguide/tsguide.html#naming-rules-by-identifier-type

sonalranjit commented on bikespace_frontend/src/components/FeedbackMailTo.tsx:
```diff
@@ -0,0 +1,41 @@
+Hi BikeSpace team,
+
+`;
+const feedback_href = [
```
Same thing as above about lower camelcase style

sonalranjit reviewed (changes requested):
Just minor formatting for variable names

tallcoleman committed:
Changed variable naming to camelCase

sonalranjit reviewed (approved):
woo thanks!
